### DISCUSSION TITLE
Rewrite in go

### DISCUSSION THREAD

**Discussion by fwartner** [fwartner/homebrew-mac-cleanup#90] [Ideas]
A question that I asked myself a couple of times before: _Does it make sense to rewrite the tool in go? **YES!**_

Features that I want to implement:

- Plugin System (So you only run what you actually need)
- Plugin-Installer (Install the plugins that you actually want)

Tell me what you think about this.. ❤️

**Comment by necrotech** [comment]
As long as the plugin system is featureful and clear to use.

For instance, I use things like this in a private copy of the script on my workstation:
`sudo find /Library/Caches -type f -atime +90 -exec rm {} \+`

That will require a good amount of complexity to implement as a configuration directive in a plugin, or plugins can just call external utilities, which eliminates some of the benefits of moving to a different language.

If you primarily want to implement things as plugins and provide the ability to run only what the end user wants, that would be fairly simple to do with the current shell version.

**Reply by fwartner** [reply]
True that.
I still need to think about it..

**Comment by efa2d19** [comment]
@fwartner
Go will be super cool, but what do you think about rewriting it in Python?
It'll be super easy to maintain by other people, Bash isn't super popular nowadays :(

Also I can do it, if you like the idea
